nemo/collections/nlp/models/text_classification/text_classification_model.py

```diff
@@ -159,31 +204,36 @@ def _setup_tokenizer(self, cfg: DictConfig):
         self.tokenizer = tokenizer
 
     def setup_training_data(self, train_data_config: Optional[DictConfig]):
-        self._train_dl = self._setup_dataloader_from_config(cfg=train_data_config)
+        self._train_dl = self._create_dataloader_from_config(cfg=train_data_config, mode='train')
 
     def setup_validation_data(self, val_data_config: Optional[DictConfig]):
-        self._validation_dl = self._setup_dataloader_from_config(cfg=val_data_config)
+        self._validation_dl = self._create_dataloader_from_config(cfg=val_data_config, mode='validation')
 
     def setup_test_data(self, test_data_config: Optional[DictConfig]):
-        self._test_dl = self._setup_dataloader_from_config(cfg=test_data_config)
+        self._test_dl = self._create_dataloader_from_config(cfg=test_data_config, mode='test')
 
-    def _setup_dataloader_from_config(self, cfg: DictConfig):
-        input_file = cfg.file_name  # os.path.join(self.dataset_cfg.data_dir, f"{cfg.file_name}")
+    def _create_dataloader_from_config(self, cfg: DictConfig, mode):
+        if not cfg or not cfg.file_path:
+            logging.info(
+                f"Dataloader config or file_path for the {mode} is missing, so no data loader for test is created!"
+            )
```

Comment:
Can we avoid adding the extra arg `mode`? If model.test_ds is not defined this method shouldn't be executed

Reply:
Actually when test loader is not set, setup_test_data is still called. Previously we have three portion of if conditions in three places with the same code to check if the file_path is null or not and give a warning. But now we have just them just on one place with no code duplicates. Should I revert it back?